EIPS/eip-7923.md

```diff
@@ -0,0 +1,283 @@
+-            # buffer so we're not missing anything by making a new buffer and forgetting
+-            # about the old one. We're keeping too much memory around but this is still
+-            # a net savings over having read() return a new bytes() object every time.
+-            self._bytes = self._bytes + bytearray(size_to_extend)
+-
+-    def __len__(self) -> int:
+-        return len(self._bytes)
++        if start_position + size >= 2**32:
++            raise PyEVMError("Non 32-bit address")
+
+-    def write(self, start_position: int, size: int, value: bytes) -> None:
+-        if size:
+-            validate_uint256(start_position)
+-            validate_uint256(size)
+-            validate_is_bytes(value)
+-            validate_length(value, length=size)
+-            validate_lte(start_position + size, maximum=len(self))
++        validate_uint256(start_position)
++        validate_uint256(size)
++        validate_is_bytes(value)
++        validate_length(value, length=size)
++
++        end_position = start_position + size
+
+-            self._bytes[start_position : start_position + len(value)] = value
++        self.memview[start_position : end_position] = value
+
++    # unused
+     def read(self, start_position: int, size: int) -> memoryview:
+-        return memoryview(self._bytes)[start_position : start_position + size]
++        return memoryview(self.memview)[start_position : start_position + size]
+
+     def read_bytes(self, start_position: int, size: int) -> bytes:
+-        return bytes(self._bytes[start_position : start_position + size])
++        return bytes(self.memview[start_position : start_position + size])
+
+     def copy(self, destination: int, source: int, length: int) -> None:
+         if length == 0:
+@@ -69,5 +68,5 @@ class Memory(MemoryAPI):
+         validate_uint256(length)
+         validate_lte(max(destination, source) + length, maximum=len(self))
+
+-        buf = memoryview(self._bytes)
++        buf = memoryview(self.memview)
+         buf[destination : destination + length] = buf[source : source + length]
+```
+
+## Security Considerations
+
+There are two primary security considerations regarding this EIP.
+
+One, does it break existing contracts? That is, do existing contracts depend on memory being restricted?
+
+Outside of gas costing, existing contracts may simply execute to completion rather than running out of gas.
+
+Two, does this enable memory-based DoS attacks against clients?
+
+This requires a maximum-memory usage analysis. Based on a gas limit today of 30mm gas, recursively calling a contract that allocates 256KB of memory in each call stack can allocate 54MB of total memory, which is not substantially different than the 64MB limit proposed in this EIP. Note that the transaction-global gas limit proposed in this EIP provides a useful invariant, which is that future changes to call stack limits will not affect the total amount of memory that can be allocated.
```

Comment:
Transaction global memory limit - right?

Reply:
yes

Reply:
clarified in 47cf03ff484b8c47390b8367dfc5ca477ac47232

Reply:
The section talks about "transaction-global gas limit", I think this should be "transaction-global memory limit". For a gas limit for the tx I'd think it would be this EIP: https://github.com/ethereum/EIPs/blob/master/EIPS/eip-7825.md

Reply:
ah, i see -- thanks for pointing that out. fixed in 4a77177a3d83543027ee71970ef992c3892abf8f